tests/storages_tests/test_journal.py

```diff
@@ -116,3 +118,21 @@ def test_pop_waiting_trial_multiprocess_safe() -> None:
                 if trial_id is not None:
                     trial_id_set.add(trial_id)
         assert len(trial_id_set) == num_enqueued
+
+
+def test_pickle_dump_and_load() -> None:
+    with tempfile.TemporaryDirectory() as workdir:
+        file_name = os.path.join(workdir, "journal.log")
+        file_storage = optuna.storages.JournalFileStorage(file_name)
+        storage = optuna.storages.JournalStorage(file_storage)
+        study = optuna.create_study(storage=storage)
+        num_enqueued = 10
+        for i in range(num_enqueued):
+            study.enqueue_trial({"i": i})
+        with open(os.path.join(workdir, "pickled.pkl"), "wb") as f:
+            pickle.dump(storage, f)
+        with open(os.path.join(workdir, "pickled.pkl"), "rb") as f:
+            loaded_storage = pickle.load(f)
```

Comment:
Do we need yo write down the pickeled object to file? It can be enough to pickle and unpickle the object.